plugins/eos/gs-plugin-eos.c

```diff
@@ -319,10 +591,25 @@ gs_plugin_setup (GsPlugin *plugin,
 	settings = g_settings_new ("org.gnome.shell");
 	priv->is_coding_enabled = g_settings_get_boolean (settings, "enable-coding-game");
 
+	g_assert (G_N_ELEMENTS (eos_updater_state_str) == EOS_UPDATER_N_STATES);
+
 	/* Synchronous, but this guarantees that the lookup table will be
 	 * there when we call ReplaceApplication later on */
 	read_icon_replacement_overrides (priv->replacement_app_lookup);
 
+	{
+		g_autoptr(GError) local_error = NULL;
+		priv->updater_proxy = eos_updater_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
```

Comment:
I wanted to make sure we have the object before a `refresh` happens. Maybe this may still be possible by doing it asynchronously, but I didn't experience any noticeable delay, and prefer to make sure we have the proxy from the beginning.
Since GNOME Software is started in the background, even if this delays things a bit, it shouldn't be noticeable to the user. I should get a better insight when I work on improving the performance, but for now I feel safer by leaving this as is.